src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -54,6 +54,10 @@ public static boolean canEncode(int cp) {
         return cp >=0 && cp <= 0xff;
     }
 
+    public static byte coderFromChar(char cp) {
+        return (byte)((0xff - cp) >>> Integer.SIZE - 1);
```

Comment:
```suggestion
        return (byte)((0xff - cp) >>> (Integer.SIZE - 1));
```
For clarity.

Or use `return (byte)((0xff - cp) >>> Character.SIZE);` as this bit is set in `0xFF - 0xFFFF`